### PR TITLE
Infrastructure: Bump eslint from 8.52.0 to 8.57.0, eslint-plugin-jsdoc from 39.3.3 to 48.2.3, eslint-plugin-html from 6.2.0 to 8.1.1, eslint-config-prettier from 8.5.0 to 9.1.0,  eslint-plugin-prettier from 4.2.1 to 5.1.3, eslint-plugin-ava from 13.2.0 to 14.0.0, and prettier from 2.7.1 to 3.2.5

### PR DESCRIPTION
**Pull request by evmiguel**
Supersedes #2943, #2995, #3009, #2891, #2596 and #2929
___
[WAI Preview Link](https://deploy-preview-304--aria-practices.netlify.app/ARIA/apg) _(Last built on Wed, 01 May 2024 14:32:06 GMT)._

### CODE REVIEW

**Comment by evmiguel** [comment]
Please hold off on reviewing this as prettier must also be updated as well.

**Comment by howard-e** [comment]
Looks like the link checker issue here should be resolved by #2965

File: content/patterns/button/examples/js/button.js [hunk 1]
```diff
@@ -24,7 +24,6 @@ function init() {
 
 /**
  * Activates the action button with the enter key.
- *
```
Comment (howard-e) [inline content/patterns/button/examples/js/button.js:27]: I think we sould preserve the current way these JSDoc lines are handled by explicitly setting a [jsdoc/tag-lines](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/tag-lines.md#tag-lines) rule.

Setting `"jsdoc/tag-lines": ["warn", "any", { "startLines":1 }]` in .eslintrc.json rules should be what we want.

File: test/util/report.js [hunk 2]
```diff
@@ -92,14 +89,13 @@ const getAttributeRowName = function ($, $tableRow) {
...
- *     missingTests: <copy of array of discovered data-test-ids>
- *     missingAttrs: <rows in attribute table missing data-test-ids>
- *     missingKeys: <rows in keyboard support table missing data-test-ids>
- *   }
+ * <exampleFile>: {
+ * existingTestIds: <array of discovered data-test-ids>
+ * missingTests: <copy of array of discovered data-test-ids>
+ * missingAttrs: <rows in attribute table missing data-test-ids>
+ * missingKeys: <rows in keyboard support table missing data-test-ids>
+ * }
```
Comment (howard-e) [inline test/util/report.js:97]: I think we want to allow contributors to preserve their own formatting in JSDoc for instances like this, so readability could be clearer. So disabling `jsdoc/check-line-alignment` should be fine.

**Review by howard-e — COMMENTED** [review]
Looks good! Thanks for grouping these like-changes together.

Since this makes many changes related to the current formatting, I spent some time looking through the new rules to see where we'd want to preserve the current behavior. I've identified 2 as inline comments. Those 2, not only because they contribute to a significant amount of changes here, which in turns gives an updated file date which isn't substantial - but also because there isn't a clear benefit of enforcing those new formatting changes (to me).

**Pull request merged**